### PR TITLE
add hash of ROS_PACKAGE_PATH to rospack/rosstack cache, remove ROS_ROOT from cache (#28)

### PR DESCRIPTION
**Pull request by dirk-thomas**
@tfoote @wjwwood Please review.

### CODE REVIEW

**Comment by wjwwood** [comment]
+1

File: src/rospack.cpp [hunk 1]
```diff
@@ -1937,10 +1920,36 @@ bool cmpDirectoryCrawlRecord(DirectoryCrawlRecord* i,
     logWarn(std::string("cannot create rospack cache directory ") +
             cache_path.string() + ": " + e.what());
   }
-  cache_path /= cache_name_;
+  cache_path /= cache_prefix_ + "_" + getCacheHash();
   return cache_path.string();
 }
 
+std::string
+Rosstackage::getCacheHash()
```
Comment (tfoote) [inline src/rospack.cpp:1929]: It would be good to use a standard hash library like boost::hash in boost functional.

Comment (dirk-thomas) [inline src/rospack.cpp:1929]: Done.

**Comment by tfoote** [comment]
+1

File: src/rospack.cpp (hunk 1 above)
Comment (wjwwood) [inline src/rospack.cpp:1929]: +1

**Pull request merged**